src/mmw/js/src/modeling/controls.js

```diff
@@ -106,6 +106,72 @@ var ConservationPracticeView = ModificationsView.extend({
     }
 });
 
+var PrecipitationSynchronizer = (function() {
+    var isEnabled,
+        precipViews = [];
+
+    // Add a view to the list of views to be kept syncrhonized
+    function add(precipView) {
+        if (isEnabled) {
+            precipViews.push(precipView);
+        }
+    }
+
+    // Remove a view from the list
+    function remove(precipView) {
+        if (isEnabled) {
+            precipViews = _.filter(precipViews, function(otherPrecipView) {
+                return (otherPrecipView !== precipView);
+            });
+        }
+    }
+
+    // Turn synchronization on
+    function on() {
+        isEnabled = true;
+        precipViews = [];
+    }
+
+    // Turn synchronization off
+    function off() {
+        isEnabled = false;
+        precipViews = [];
+    }
+
+    // Synchronize the group to the given slider
+    function syncTo(precipView) {
+        var value = precipView.ui.slider.val();
+
+        if (isEnabled) {
+            isEnabled = false;
+            precipViews.forEach(function(otherPrecipView) {
+                var otherValue = otherPrecipView.ui.slider.val();
+                if (otherValue !== value) {
+                    otherPrecipView.ui.slider.val(value);
+                    otherPrecipView.onSliderChanged();
+                }
+            });
+            isEnabled = true;
+        }
+    }
+
+    // Synchronize the group to the first slider
+    function sync() {
+        if (precipViews.length > 0) {
+            syncTo(precipViews[0]);
+        }
+    }
+
+    return {
```

Comment:
I really like this way of returning functions defined elsewhere. Makes the API very easy to grok.